BJS2-26960 Отправка SMS сообщений

Murzin34 opened the pull request:
В application.yaml вместо some_api_key необходимо вставить API-key MTS Exolve (там только API-key, без secret), и вместо поля some_sender_number вставить номер телефона который предоставляет сервис MTS Exolve.

Murzin34 committed:
BJS2-26960 Отправка SMS сообщений

Murzin34 committed:
Фикс.
Перенес параметры запроса и хедеры в ямл.

Murzin34 committed:
Фикс.

Murzin34 committed:
Фикс.

Murzin34 committed:
Merge remote-tracking branch 'origin/feature-BJS2-26960' into feature-BJS2-26960

Murzin34 committed:
Merge branch 'kraken-master-stream6' into feature-BJS2-26960

Murzin34 committed:
Небольшой фикс.

DankeVilletti commented on src/main/java/faang/school/notificationservice/service/SmsNotificationService.java:
```diff
@@ -0,0 +1,30 @@
+
+    @Override
+    public void send(UserDto user, String message) {
+        HttpClient client = smsNotificationServiceHandler.getHttpClient();
+        HttpRequest request = smsNotificationServiceHandler.getHttpRequest(user, message);
+
+        smsNotificationServiceHandler.retryableSend(client, request);
+    }
```
ты разделил ответственность клиента(запроса) на два метода (smsNotificationServiceHandler.retryableSend(client, request))
Лучше всю логику HTTP вынести в один метод или оставить здесь, но повесить @Retryable

Murzin34 replied:
Разделил т.к. конфиг по получению HttpClient и построению HttpRequest могут быть различны.

Разве метод который может по несколько раз создавать, и новый HttpClient, и новый HttpRequest, и отправлять сообщение (в зависимости от конфига `@Retryable`) будет лучше?

Пока оставлю как есть.

DankeVilletti commented on src/main/java/faang/school/notificationservice/service/sms/SmsNotificationServiceHandler.java:
```diff
@@ -0,0 +1,65 @@
+                .build();
+    }
+
+    @Retryable(retryFor = {MtsExolveException.class}, maxAttempts = 2, backoff = @Backoff(delay = 10000))
```
Лучше вынести в yaml

Murzin34 replied:
О и правда, спасибо)

DankeVilletti commented on src/main/java/faang/school/notificationservice/service/sms/SmsNotificationServiceHandler.java:
```diff
@@ -0,0 +1,65 @@
+    public void retryableSend(HttpClient client, HttpRequest request) {
+        try {
+            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+            log.info("Response status code: " + response.statusCode());
```
Лучше одним логом, они могут быть разорваны многопоточкой насколько я понимаю

Murzin34 replied:
Действительно лучше перестраховаться) Спасибо)

DankeVilletti commented on src/main/java/faang/school/notificationservice/service/sms/MtsExolveProperties.java:
```diff
@@ -0,0 +1,33 @@
+
+@Getter
+@Component
+public class MtsExolveProperties {
```
нэйминг, наверное лучше API добавить для понимания

Murzin34 replied:
По итогу переименовал этот класс в SmsNotificationProperties, а mts_exolve_api использую в application.yaml

DankeVilletti reviewed (commented):
воть

Murzin34 committed:
Фикс после кросс-ревью.
Переименовал MtsExolveProperties в SmsNotificationProperties, добавил поля в application.yaml, поправил SmsNotificationServiceHandler и тесты.

Murzin34 committed:
Фикс.
Добавил поле в CommentEventDto

20cognosce commented on src/main/java/faang/school/notificationservice/service/sms/SmsNotificationProperties.java:
```diff
@@ -0,0 +1,39 @@
+
+@Getter
+@Component
+public class SmsNotificationProperties {
```
@ConfigurationProperties(prefix) переделать

20cognosce reviewed (approved):
Хорошо, жду демо
